grails-app/services/au/org/emii/portal/JsonService.groovy

```diff
@@ -0,0 +1,46 @@
+package au.org.emii.portal
+
+
+import groovyx.net.http.HTTPBuilder
+import groovyx.net.http.HttpResponseException
+
+import static groovyx.net.http.ContentType.JSON
+import static groovyx.net.http.Method.POST
+
+class JsonService extends AsyncDownloadService {
+
+    def groovyPageRenderer
+    def grailsApplication
+
+    String registerJob(params) throws HttpResponseException {
+        getConnection(params).request(POST) { req ->
+            body = getBody(params)
+            response.success = onResponseSuccess
+        }
+    }
+
+    def getConnection(params) {
+        def conn = new HTTPBuilder(params.server)
+
+        //  If an X-Forwarded-For param was passed - set it as a HTTP parameter
+        if(params["X-Forwarded-For"] != null) {
```

Comment:
Can we separate normal request parameters from request headers so we don't have to test for individual headers to include?

Reply:
Also, not really clear what this is for - perhaps explain why its needed in a comment somewhere